File: frontend/user/src/api/workflow.js
// 工作流 API —— 新建文件：frontend/user/src/api/workflow.js
// 复用项目已有的 createApi() 封装工厂（与 format_check.js、knowledge.js 同一套体系）
import { createApi } from './config.js'

const api = createApi()

// ========== 模板（写死在后端常量，直接返回） ==========
export const listWorkflowTemplates = () => api.get('/workflow/templates')

// ========== 实例 ==========
export const listWorkflowInstances = (params) => api.get('/workflow/instances', { params })
export const createWorkflowInstance = (data) => api.post('/workflow/instances', data)
export const getWorkflowInstance = (id) => api.get(`/workflow/instances/${id}`)
export const updateWorkflowInstance = (id, data) => api.put(`/workflow/instances/${id}`, data)
export const deleteWorkflowInstance = (id) => api.delete(`/workflow/instances/${id}`)

// ========== 节点 ==========
export const updateWorkflowNode = (nodeInstId, data) => api.put(`/workflow/nodes/${nodeInstId}`, data)
export const generateWorkflowNode = (nodeInstId, instruction = '', save = true) =>
  api.post(`/workflow/nodes/${nodeInstId}/generate`, { instruction, save })

// ========== 会议核心上下文（基础信息节点的 AI 入口） ==========
export const parseNaturalLanguage = (instId, text) =>
  api.post(`/workflow/instances/${instId}/parse-natural-language`, { text })
export const parseKeyValue = (instId, text) =>
  api.post(`/workflow/instances/${instId}/parse-key-value`, { text })
export const confirmWorkflowContext = (instId, workflowContext, confirmOverrides = {}) =>
  api.post(`/workflow/instances/${instId}/confirm-context`,
    { workflow_context: workflowContext, confirm_overrides: confirmOverrides })

File: server/app/interfaces/workflow.py
"""工作流路由。路径与响应结构与旧系统一致（/workflow/...）。"""
from typing import Optional

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.application.shared.writing_assistant import WritingAssistant
from app.application.workflow.dto import (
    ConfirmContextRequest,
    CreateInstanceRequest,
    GenerateNodeRequest,
    ParseKeyValueRequest,
    ParseNaturalLanguageRequest,
    UpdateInstanceRequest,
    UpdateNodeRequest,
)
from app.application.workflow.service import WorkflowService
from app.domain.identity.entities import User
from app.infrastructure.ai import get_llm_gateway
from app.infrastructure.database import get_db
from app.interfaces.deps import get_current_user

router = APIRouter(prefix="/workflow", tags=["workflow"])


def get_workflow_service(db: Session = Depends(get_db)) -> WorkflowService:
    return WorkflowService(db, WritingAssistant(get_llm_gateway()))


@router.get("/templates")
def list_templates(user: User = Depends(get_current_user),
                   svc: WorkflowService = Depends(get_workflow_service)):
    return svc.list_templates()


@router.get("/instances")
def list_instances(status: Optional[str] = Query(None),
                   user: User = Depends(get_current_user),
                   svc: WorkflowService = Depends(get_workflow_service)):
    return svc.list_instances(user, status)


@router.post("/instances")
def create_instance(req: CreateInstanceRequest,
                    user: User = Depends(get_current_user),
                    svc: WorkflowService = Depends(get_workflow_service)):
    return svc.create_instance(user, req.template_code, req.title, req.workflow_context)


@router.get("/instances/{inst_id}")
def get_instance(inst_id: str, user: User = Depends(get_current_user),
                 svc: WorkflowService = Depends(get_workflow_service)):
    return svc.get_instance(inst_id)


@router.put("/instances/{inst_id}")
def update_instance(inst_id: str, req: UpdateInstanceRequest,
                    user: User = Depends(get_current_user),
                    svc: WorkflowService = Depends(get_workflow_service)):
    return svc.update_instance(inst_id, req.title, req.status,
                               req.basic_info, req.workflow_context)


@router.delete("/instances/{inst_id}")
def delete_instance(inst_id: str, user: User = Depends(get_current_user),
                    svc: WorkflowService = Depends(get_workflow_service)):
    return svc.delete_instance(inst_id)


@router.post("/instances/{inst_id}/parse-natural-language")
def parse_natural_language(inst_id: str, req: ParseNaturalLanguageRequest,
                           user: User = Depends(get_current_user),
                           svc: WorkflowService = Depends(get_workflow_service)):
    return svc.parse_natural_language(inst_id, req.text)


@router.post("/instances/{inst_id}/parse-key-value")
def parse_key_value(inst_id: str, req: ParseKeyValueRequest,
                    user: User = Depends(get_current_user),
                    svc: WorkflowService = Depends(get_workflow_service)):
    return svc.parse_key_value(inst_id, req.text)


@router.post("/instances/{inst_id}/confirm-context")
def confirm_context(inst_id: str, req: ConfirmContextRequest,
                    user: User = Depends(get_current_user),
                    svc: WorkflowService = Depends(get_workflow_service)):
    return svc.confirm_context(inst_id, req.workflow_context, req.confirm_overrides or {})


@router.put("/nodes/{node_inst_id}")
def update_node(node_inst_id: str, req: UpdateNodeRequest,
                user: User = Depends(get_current_user),
                svc: WorkflowService = Depends(get_workflow_service)):
    return svc.update_node(node_inst_id, req.content, req.status)


@router.post("/nodes/{node_inst_id}/generate")
def generate_node(node_inst_id: str, req: GenerateNodeRequest,
                  user: User = Depends(get_current_user),
                  svc: WorkflowService = Depends(get_workflow_service)):
    return svc.generate_node(node_inst_id, req.instruction, req.save)
